### PR TITLE
Clean up non-ANSI displays for CHAINS

### LINKED ISSUE #1529: Chains displays high-bit chars to non-ANSI users
cosmetic bug. change hi-bit lines to ASCII output.

### PR DESCRIPTION
**Pull request by ericpareja**
Don't display hi-bit line chars to non-ANSI users. Replace with ASCII chars.
Fixes #1529

### CODE REVIEW

File: bbs/chains.cpp [hunk 1]
```diff
@@ -121,37 +121,60 @@ static void show_chains(int *mapp, std::map<int, int>& map) {
...
+				 i + 1, bar,
+				 a()->chains->at(map[i]).description, bar,
+				 a()->chains->at(map[i]).usage, bar), &abort);
 	}
       }
     }
+      if (okansi()) {
     bout.outstr("|#7\xC0\xC4\xC4\xC1\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC1\xC4\xC4\xC4\xC4\xC4\xC1\xC4\xC4\xC1\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC1\xC4\xC4\xC4\xC4\xC4\xD9\r\n");
+      } else {
+	bout.bpla(fmt::sprintf("+--+----------------------------+-----+--+----------------------------+-----+"), &abort);
```
Comment (wwiv) [inline bbs/chains.cpp]: there are no params here, remove the fmt::sprintf wrapper

File: bbs/chains.cpp [hunk 2]
```diff
@@ -121,37 +121,60 @@ static void show_chains(int *mapp, std::map<int, int>& map) {
...
+      }
     } else {
-    bout.outstr("|#7\xDA\xC4\xC4\xC2\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC2\xC4\xC4\xC2\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xBF\r\n");
+      if (okansi()) {
+	bout.outstr("|#7\xDA\xC4\xC4\xC2\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC2\xC4\xC4\xC2\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xBF\r\n");
+      } else {
+	bout.bpla(fmt::sprintf("+--+----------------------------------+--+----------------------------------+"), &abort);
```
Comment (wwiv) [inline bbs/chains.cpp]: remove sprintf (again)

File: bbs/chains.cpp [hunk 3]
```diff
@@ -121,37 +121,60 @@ static void show_chains(int *mapp, std::map<int, int>& map) {
...
+      }
+    auto bar = fmt::sprintf("|#%d%s", FRAME_COLOR, okansi() ? "\xB3": "|") ;
```
Comment (wwiv) [inline bbs/chains.cpp]: please use format
"|#{:d}{}" woud be right to use.  The :d just to make sure it treats it as an int

File: bbs/chains.cpp [hunk 4]
```diff
@@ -121,37 +121,60 @@ static void show_chains(int *mapp, std::map<int, int>& map) {
     bout.litebar(StrCat(a()->config()->system_name(), " Online Programs"));
     if (a()->config()->toggles().show_chain_usage) {
       bout.bpla(fmt::sprintf(" |#5Num |#1%-28.28s|#1%-5.5s |#5Num |#1%-28.28s|#1%-5.5s", "Description", "Usage", "Description", "Usage"), &abort);
+      if (okansi()) {
       bout.outstr("|#7\xDA\xC4\xC4\xC2\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC2\xC4\xC4\xC4\xC4\xC4\xC2\xC4\xC4\xC2\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC4\xC2\xC4\xC4\xC4\xC4\xC4\xBF\r\n");
+      } else {
+	bout.bpla(fmt::sprintf("+--+----------------------------+-----+--+----------------------------+-----+"), &abort);
```
Comment (wwiv) [inline bbs/chains.cpp]: remove the fmt:sprintf, there's no parameters here

File: bbs/chains.cpp [hunk 5]
```diff
@@ -121,37 +121,60 @@ static void show_chains(int *mapp, std::map<int, int>& map) {
...
+      }
     for (auto i = 0; i < *mapp && !abort && !a()->sess().hangup(); i++) {
-      bout.outstr(fmt::sprintf("|#7\xB3|#2%2d|#7\xB3 |#1%-27.27s|#7\xB3|#1%5d|#7\xB3", i + 1, a()->chains->at(map[i]).description, a()->chains->at(map[i]).usage), &abort, &next);
+      auto bar = fmt::sprintf("|#%d%s", FRAME_COLOR, okansi() ? "\xB3": "|") ;
```
Comment (wwiv) [inline bbs/chains.cpp]: please add const

File: bbs/chains.cpp [hunk 6]
```diff
@@ -121,37 +121,60 @@ static void show_chains(int *mapp, std::map<int, int>& map) {
...
+      }
+    auto bar = fmt::sprintf("|#%d%s", FRAME_COLOR, okansi() ? "\xB3": "|") ; 
     for (auto i = 0; i < *mapp && !abort && !a()->sess().hangup(); i++) {
-      bout.outstr(fmt::sprintf("|#7\xB3|#2%2d|#7\xB3 |#1%-33.33s|#7\xB3", i + 1, a()->chains->at(map[i]).description), &abort, &next);
+      bout.outstr(fmt::sprintf("%s|#2%2d%s |#1%-33.33s%s", bar, i + 1, bar, a()->chains->at(map[i]).description, bar), &abort, &next);
       i++;
       if (!abort && !a()->sess().hangup()) {
         if (i >= *mapp) {
-          bout.bpla(fmt::sprintf("  |#7\xB3                                  |#7\xB3"), &abort);
+          bout.bpla(fmt::sprintf("  %s                                  %s",bar,bar), &abort);
```
Comment (wwiv) [inline bbs/chains.cpp]: please use fmt::format

**Review by wwiv — CHANGES_REQUESTED** [review]

File: bbs/chains.cpp (hunk 5 above)
Comment (ericpareja) [reply bbs/chains.cpp:130]: auto const?

**Review by ericpareja — COMMENTED** [review]
Tidied up the other occurrences of fmt::sprintf() to fmt::format() and tested build on Debian 10.13.

**Review by wwiv — APPROVED** [review]

**Pull request merged**